cloudinit/distros/__init__.py

```diff
@@ -983,6 +984,40 @@ def do_as(self, command: list, user: str, cwd: str = "", **kwargs):
             **kwargs,
         )
 
+    @property
+    def is_virtual(self) -> bool:
+        """Detect if is running as a virtual machine or bare metal.
+        If fails, it returns True to avoid breaking changes.
```

Comment:
IMO this is going to be confusing and misleading. What about encoding the unknown case as None?